### PR TITLE
OM-44460 Add CPURequestAllocation and MemoryRequestAllocation into VDC

### PR DESCRIPTION
**Pull request by ading1977**
The code change is big, so I have splitted them into multiple commits. Please review each individual commit.

### CODE REVIEW

File: pkg/discovery/repository/kube_cluster.go [hunk 1]
```diff
@@ -176,21 +173,23 @@ func NewKubeNode(apiNode *v1.Node, clusterName string) *KubeNode {
...
 
 	// Node compute resources and properties
-	nodeEntity.UpdateResources(apiNode)
+	nodeEntity.updateResources(apiNode)
 	return nodeEntity
 }
 
 // Set the compute resources and IP property in the node entity
-func (nodeEntity *KubeNode) UpdateResources(apiNode *v1.Node) {
+func (nodeEntity *KubeNode) updateResources(apiNode *v1.Node) {
```
Comment (enlinxu) [inline pkg/discovery/repository/kube_cluster.go]: Why do we want to change here? I thought UpdateResources is the right convention?

Comment (ading1977) [reply pkg/discovery/repository/kube_cluster.go]: `updateResources` is only called within the package. We are not exporting it to be used by any other package, so we should use lower case according to [Go language specification](https://golang.org/ref/spec#Exported_identifiers)

**Comment by ading1977** [comment]
Format changes are checked with gofmt.
We cannot use golint as there are too many style mistakes in our existing code base :-(, that will require a separate effort to correct them.

File: pkg/discovery/repository/kube_cluster.go (hunk 1 above)
Comment (enlinxu) [reply pkg/discovery/repository/kube_cluster.go]: I see, I wasn't aware of it. Then I am good with the review. Ship it!

**Pull request merged**